scripts/ci/generate_install_lists.py

```diff
@@ -84,7 +84,13 @@ def main(argv=sys.argv[1:]):
         for package_root in args.package_root[0:-1]:
             print("Crawling for packages in '%s'" % package_root)
             underlay_pkgs.update(find_packages(package_root))
-            all_underlay_pkg_names |= locate_packages(package_root).keys()
+
+            # Check for a colcon index for non-ROS package detection
+            colcon_index = os.path.join(package_root, 'colcon-core', 'packages')
```

Comment:
I am wondering why `package_root` points to the `share` directory?

Reply:
This might be a victim of over-reusing the common argument adding methods. Here in the `generate_install_lists.py` script, `package_root` means something more like "package search path".

Here is where those search paths are computed: https://github.com/ros-infrastructure/ros_buildfarm/blob/master/ros_buildfarm/templates/ci/create_workspace.Dockerfile.em#L85

I can rename the argument to make it more clear if you like.

Reply:
No need to do it in this PR. I was just partly confused and partly curious.